lib/api/completeMultipartUpload.js

```diff
@@ -467,19 +311,23 @@ function completeMultipartUpload(authInfo, request, log, callback) {
                     // in cases where completing mpu overwrites a previous
                     // null version when versioning is suspended or versioning
                     // is not enabled, need to delete pre-existing data
+                    // unless the preexisting object and the completed mpu
+                    // are on Azure data backend
```

Comment:
because we don't allow object puts/initiate mpu to azure when a bucket has versioning enabled? if the possibility is blocked, is this needed? if it's not blocked, shouldn't it be until we are able to have azure versioning on versioning?

Reply:
no, this dataToDelete is only done when versioning is *not* enabled. since we don't have versioning on Azure, this call to batchDelete would delete the completed MPU object on Azure. We don't encounter this on AWS because versioning is always enabled.